Extension/src/LanguageServer/ui_new.ts

```diff
@@ -403,6 +416,26 @@ export class NewUI implements UI {
         }
     }
 
+    private compilerTimout?: NodeJS.Timeout;
+    public async showCompilerStatusIcon(show: boolean): Promise<void> {
+        if (!telemetry.showStatusBarIntelliSenseIndicator()) {
+            return;
+        }
+        if (this.compilerTimout) {
+            clearTimeout(this.compilerTimout);
+        }
+        if (show) {
+            this.compilerTimout = setTimeout(() => {
+                if (show) {
```

Comment:
This bool is passed by value -- it needs to be re-evaluated, i.e. via either copying the root checks or passing a lambda function.

Reply:
Can you give a code snippet?